packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -154,11 +186,144 @@ async function createOutgoingPayment(
       }
     } else if (isOutgoingPaymentError(err)) {
       return err
+    } else if (err instanceof KnexTimeoutError) {
+      return OutgoingPaymentError.GrantLocked
```

Comment:
We're generally returning `OutgoingPaymentError` for caller related issues (invalid options, etc).
I'm thinking this should log and still throw.